governance/xc_admin/packages/xc_admin_frontend/pages/index.tsx

```diff
@@ -13,26 +14,28 @@ import { StatusFilterProvider } from '../contexts/StatusFilterContext'
 import { classNames } from '../utils/classNames'
 import '../mappings/signers.json'
 
+const readPublisherKeyToNameMapping = (filename: string) => {
+  if (fs.existsSync(filename)) {
+    return YAML.parse(fs.readFileSync(filename, 'utf8')).reduce(
+      (acc: { [key: string]: string }, rec: { key: string; name: string }) => {
```

Comment:
Note that the type casting here removes typescript's ability to actually guarantee anything about the types.  IMO this shouldn't even exist in typescript.

Typically any time I load data from an external source I use [zod](https://zod.dev/) to parse the data so I can guarantee that my typescript types are actually correct and can handle invalid data accordingly, rather than typecasting and assuming the types are accurate.

Reply:
yeah I know... the original code didn't have any typechecking either so I put in the cast to shut the compiler up. Since this is a file that we control (and that at least 2 other projects will use) I think it's fairly safe to do this, WDYT?

Reply:
as a side note I also think the amount of things people do in typescript to silence the compiler without actually fixing the underlying type safety issues is a huge problem and it's a big reason why typescript isn't nearly as reliable as something like rust.... good discipline pays off IMHO.

Reply:
I *could* be persuaded to add some rudimentary validation and throwing exceptions (or `os.exit(1)`) on failure.

Reply:
You could do that, but it will be more work and less flexible than just using zod.  We need zod for this app anyways so it wouldn't be like it's a dependency I don't plan to introduce regardless.